crates/file_finder/src/file_finder.rs

```diff
@@ -230,88 +274,104 @@ impl Matches {
         new_search_matches: impl Iterator<Item = ProjectPanelOrdMatch>,
         extend_old_matches: bool,
     ) {
-        let no_history_score = 0;
-        let matching_history_paths =
-            matching_history_item_paths(history_items.clone(), currently_opened, query);
-        let new_search_matches = new_search_matches
-            .filter(|path_match| !matching_history_paths.contains_key(&path_match.0.path))
-            .map(Match::Search)
-            .map(|m| (no_history_score, m));
-        let old_search_matches = self
-            .matches
-            .drain(..)
-            .filter(|_| extend_old_matches)
-            .filter(|m| matches!(m, Match::Search(_)))
-            .map(|m| (no_history_score, m));
-        let history_matches = history_items
-            .into_iter()
-            .chain(currently_opened)
-            .enumerate()
-            .filter_map(|(i, history_item)| {
-                let query_match = matching_history_paths
-                    .get(&history_item.project.path)
-                    .cloned();
-                let query_match = if query.is_some() {
-                    query_match?
-                } else {
-                    query_match.flatten()
-                };
-                Some((i + 1, Match::History(history_item.clone(), query_match)))
-            });
+        self.opened_path = currently_opened.cloned();
+
+        let Some(query) = query else {
+            //NOTE: assuming that if there's no query, then there's no search matches.
+            self.matches.clear();
+            let path_to_entry = |found_path: &FoundPath| Match::History {
+                path: found_path.clone(),
+                panel_match: None,
+            };
+            //NOTE: add currently_opened first, if anything is opened
```

Comment:
This and the next note seems quite obviously written in the code already, let's remove them.